metrics/firebase/docs/features/builds_aggregation/01_firestore_builds_aggregation.md

```diff
@@ -1,39 +1,169 @@
+
+For now, to make builds aggregations we should load all builds and process calculations on the client, because Cloud Firestore does not support the aggregation queries.
+
+This solution is okay if a project has not many builds, but over time, the number of builds growth and it will pull along increase the number of reads from the database and leads to heavy load/price implications.
+
+### Requirements
+> Define requirements and make sure that they are complete.
+
+- Reduces Firestore usage.
+- Processes aggregation calculations.
+- Works correctly in case of concurrent document edits.
+
+### Landscape
+> Look for existing solutions in the area.
+
+There are a few approaches to get data for the aggregated metrics:
+- Load all collection's documents and provide aggregation using the client.
+- Aggregate data in the Firestore Cloud Functions using:
+  - `transaction`;
+  - `FieldValue.increment`;
+  - `distributed counter`.
+
+Let's take a closer look at them.
+
+#### Loading all documents
+
+At this time, for the aggregated metric `builds per week`, we download all the builds of the project for the last 7 days using Firebase advanced query, count the number of builds, and display the information on the `Metrics` dashboard.
+
+Pros:
+- Easy to implement.
+
+Cons:
+- With an increase in the number of builds, the number of reads from the database grows respectively. It leads to heavy load/price implications.
+- Increases the time of calculating the aggregation metric.
+
+#### Aggregating data using the Cloud Functions
+
+With Cloud Functions, we can move our aggregation logic to the cloud and process calculations on the back-end, provided by Firebase.
+
+There are several ways, we can use the aggregation with the Cloud Functions - using the [Firebase transactions](#firebase-transactions), the [FieldValue.increment](#fieldValue.increment) method or create a [Distributed counter](#distributed-counter).
+
+#### Firebase transactions
+
+The transaction is a set of read and write operations on one or more documents. Transactions are useful when we want to update a field's value based on its current value, or the value of some other field.
+
+Pros:
+- Code runs in the cloud, so the client is less loaded.
+- In the case of a concurrent edit, the entire transaction runs again.
+
+Cons:
+- There is a maximum count of transaction reruns, that's why some transactions may fail.
+- An extra read a field's value before updating.
+- Has a maximum writes and time [limits](https://firebase.google.com/docs/firestore/quotas#writes_and_transactions).
+- Requires a lot of code to implement a simple logic(e.g.g. counters).
```

Comment:
Please check all occurrences in this document - not sure the `e.g.g.` is right if "for example" is meant (`e.g.`). Also, it seems like it should be separated by a comma (https://www.grammarly.com/blog/know-your-latin-i-e-vs-e-g/).
```suggestion
- Requires a lot of code to implement a simple logic(e.g., counters).
```